org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/resource/Storage2UriMapperJavaImpl.java

```diff
@@ -669,6 +669,7 @@ protected void doInitializeCache() throws CoreException {
 				@Override
 				public void run(IProgressMonitor monitor) throws CoreException {
 					if(!isInitialized) {
+						JavaCore.initializeAfterLoad(monitor);
```

Comment:
can this lead to deadlocks?

Reply:
I would avoid calling internal init of JDT here, that is really not supposed to be done by 3rd party code.
Sounds like a hack.